docs/docs/guide/03-hello.md

```diff
@@ -118,6 +118,236 @@ name: Hello, bob!
+				return err
+			}
+			return clientCtx.PrintProto(res)
+		},
+	}
+	flags.AddQueryFlagsToCmd(cmd)
+	return cmd
+}
+```
+
+The code defines a `CmdSayHello` command. The command is defined using the
+`cobra` library, which is a popular framework for building command-line
+applications in Go. The command accepts a `name` as an argument and uses it to
+create a `QuerySayHelloRequest` struct that is passed to the `SayHello` function
+from the `types.QueryClient`. The `SayHello` function is used to send the
+`say-hello` query to the blockchain, and the response is stored in the `res`
+variable.
+
+The `QuerySayHelloRequest` struct is defined in the `query.proto` file, which is
+a Protocol Buffer file that defines the request and response types for the
+query. The `QuerySayHelloRequest` struct includes a `Name` field of type
+`string`, which is used to provide the name to be included in the response
+message.
+
+After the query has been sent and the response has been received, the code uses
+the `clientCtx.PrintProto` function to print the response to the console. The
+`clientCtx` variable is obtained using the `client.GetClientQueryContext`
+function, which provides access to the client context, including the client's
+configuration and connection information. The `PrintProto` function is used to
+print the response using the Protocol Buffer format, which allows for efficient
+serialization and deserialization of the data.
+
+The `flags.AddQueryFlagsToCmd` function is used to add query-related flags to
+the command. This allows users to specify additional options when calling the
+command, such as the node URL and other query parameters. These flags are used
+to configure the query and provide the necessary information to the `SayHello`
+function, allowing it to connect to the blockchain and send the query.
+
+To make the `CmdSayHello` command available to users, you will need to add it to
+the chain's binary. This is typically done by modifying the
+`x/hello/client/cli/query.go` file and adding the
+`cmd.AddCommand(CmdSayHello())` statement. This adds the `CmdSayHello` command
+to the list of available commands, allowing users to call it from the
+command-line interface (CLI).
+
+```go title="x/hello/client/cli/query.go"
+func GetQueryCmd(queryRoute string) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:                        types.ModuleName,
+		Short:                      fmt.Sprintf("Querying commands for the %s module", types.ModuleName),
+		DisableFlagParsing:         true,
+		SuggestionsMinimumDistance: 2,
+		RunE:                       client.ValidateCmd,
+	}
+	cmd.AddCommand(CmdQueryParams())
+	// highlight-next-line
+	cmd.AddCommand(CmdSayHello())
+	return cmd
+}
+```
+
+Once you have provided a CLI command, users will be able to call the `say-hello`
+query and receive the appropriate response. For example, they can use the
+following command to submit the query and receive the response:
+
+```
+hellod q hello say-hello bob
```

Comment:
This will fail because the chain isn't running. 

I understand the purpose of this tutorial, and I think it's a great idea, but just as we used `ignite` to scaffold the chain, we should use `ignite` also here to run it. Unless it's possible with `appd start`, but I think it will be more complicated.

Reply:
I agree, I think at this point in users' journey we should still use `ignite chain serve` to start the chain.

Reply:
https://github.com/ignite/cli/pull/3227/commits/4c4bea3f922934eef00c2713282910f33afd146d